### ISSUE TITLE
help/version/etc. aliases are reset by `help()`/`version()`/etc.

### ISSUE THREAD

**Issue by JC3** [yargs/yargs#2203]
yargs 17.5.1

if an alias for "help" is added after `help()` is specified, the alias takes effect as expected. if the alias is added before `help()` is specified, the alias is ignored. the same applies to `version()`. the behavior is not consistent with the behavior of user-defined options.

**example**

```
    yargs(['-h'])
        .alias('help', 'h')
        .help()
        .strict()
        .argv;
```

**expected**: success with help message
**actual**: failure due to unknown option -h

**more complete example**

```
import yargs from 'yargs';

let SECTION = msg => console.log(`\n======${msg?` ${msg} `:''}======\n`);

// succeeds when alias() is after help()
SECTION('help -> alias');
yargs(['-h'])
    .help()
    .alias('help', 'h')
    .exitProcess(false)
    .strict()
    .argv;

SECTION('alias -> help');
try {
    // fails when alias() is before help()
    yargs(['-h'])
        .alias('help', 'h')
        .help()
        .exitProcess(false)
        .strict()
        .argv;
} catch { }

SECTION('sanity check');
yargs(['--help'])
    .alias('combust', 'c')
    .option('combust', {})
    .option('invert', {})
    .alias('invert', 'i')
    .exitProcess(false)
    .strict()
    .argv;
```

output:

```
====== help -> alias ======

Options:
      --version  Show version number                                   [boolean]
  -h, --help     Show help                                             [boolean]

====== alias -> help ======

Options:
  --version  Show version number                                       [boolean]
  --help     Show help                                                 [boolean]

Unknown argument: h

====== sanity check ======

Options:
      --help     Show help                                             [boolean]
  -c, --combust
  -i, --invert
```

**Comment by bcoe** [comment]
@JC3 help is enabled by default, so I believe there's a chance that explicitly calling `.help()` resets this behaviour.

tldr; I think this is working as intended.

**Comment by bcoe** [comment]
>  i just noticed the third output above is missing --version from the options list. dunno if that's a bug or not.

Quite possibly a bug, if you want to open an issue with a minimal reproduction 👍

**Comment by julianpoemp** [comment]
this is still an issue. "-h" shows help as expected, "--help" shows only help for version and help.

**Comment by shadowspawn** [comment]
@julianpoemp I think you have a different problem than what this issue is about. Please open a new issue with some example code.
